core/kazoo_documents/src/kzd_ledgers.erl

```diff
@@ -255,19 +259,19 @@ set_account_name(Doc, AccountName) ->
 %% @doc
 %% @end
 %%------------------------------------------------------------------------------
--spec unit_amount(doc()) -> kz_term:api_integer().
+-spec unit_amount(doc()) -> kz_term:api_number().
 unit_amount(Doc) ->
     unit_amount(Doc, 'undefined').
 
 -spec unit_amount(doc(), Default) -> kz_currency:units() | Default.
 unit_amount(Doc, Default) ->
-    kz_json:get_integer_value(?AMOUNT, Doc, Default).
+    kz_json:get_number_value(?AMOUNT, Doc, Default).
```

Comment:
@jamesaimonetti why number value here?  this should be in units and units should only be integers.

Reply:
Yeah, I didn't realize you were changing how the kzd modules work - generally they map accessor functions to schema fields. So when I saw `<<"amount">>` i changed it to be number instead of integer (and was curious why you called it unit_amount at first.

The trouble is, this module now breaks expectations when using it because we can't fetch "amount" and expect it to be the schema's definition of "amount" - it is now overridden to represent the "units".